### PR TITLE
Prohibit root from adding workflowitems

### LINKED ISSUE #380: ui/api: Prohibit root from adding workflowitems
Follow up to #39

### PR DESCRIPTION
**Pull request by miamarina**
# Changes

* ui: disable create workflow item button for root
* api: return precondition error if root tries to add a workflow item
* add tests

# How to test

Tests workflowitem_create.spec.ts and workflowitem_edit_spec.js are added

Closes #380

### CODE REVIEW

File: api/src/service/domain/workflow/workflowitem_create.spec.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      displayName: "test",
+      amountType: "N/A",
+    };
+
+    const result = await WorkflowitemCreate.createWorkflowitem(ctx, user, data, {
+      workflowitemExists: async (_projectId, _subprojectId, _workflowitemId) => false,
+      getSubproject: async () => baseSubproject,
+    });
+
+    assert.isOk(Result.isErr(result));
```
Comment (mathiashoeld) [inline api/src/service/domain/workflow/workflowitem_create.spec.ts]:
```suggestion
    assert.isTrue(Result.isErr(result));
```

File: api/src/service/domain/workflow/workflowitem_create.spec.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  });
```
Comment (mathiashoeld) [inline api/src/service/domain/workflow/workflowitem_create.spec.ts]: add `assert.instanceOf(...)`

File: api/src/service/domain/workflow/workflowitem_create.spec.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
+import { assert } from "chai";
+
+import { Ctx } from "../../../lib/ctx";
+import * as Result from "../../../result";
+import { ServiceUser } from "../organization/service_user";
+import * as WorkflowitemCreate from "./workflowitem_create";
+import { Subproject } from "./subproject";
```
Comment (Stezido) [inline api/src/service/domain/workflow/workflowitem_create.spec.ts]: organize imports

Comment (mathiashoeld) [reply api/src/service/domain/workflow/workflowitem_create.spec.ts]: hint: use the VS Code extension `typescript hero` to do this automatically

**Pull request merged**